### ISSUE TITLE
IndentationCheck: Incorrect line wrap handling.

### ISSUE THREAD

**Issue by alex-zuy** [checkstyle/checkstyle#766]
IndentationCheck handles line wrap incorrectly. 
Following code samples taken from test-inputs.
[if](https://github.com/pirat9600q/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/indentation/InputValidIfIndent.java#L220)
```java
if (test //indent:8 exp:8
) { //indent:8 exp:8 <- sholud be 12
    System.getProperty("blah"); //indent:12 exp:12
} //indent:8 exp:8
```
[for](https://github.com/pirat9600q/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/indentation/InputValidForIndent.java#L77)
```java
for (int i=0; //indent:8 exp:8
    i<10; i++ //indent:12 exp:12
) { //indent:8 exp:8 <- should be 12
System.getProperty("foo"); //indent:12 exp:12
} //indent:8 exp:8
```
[method call](https://github.com/pirat9600q/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/indentation/InputValidMethodIndent.java#L148)
```java
System.out.println("methods are: " + Arrays.asList( //indent:8 exp:8
    new String[] {"method"} //indent:12 exp:>=12
).toString()); //indent:8 exp:8 <- should be 12
```
and so on.

**Comment by romani** [comment]
it looks like that "(" and ")" are treated in the same way as "{" "}".
close brace are always on the same indentation as line where is open brace .

Please give me a call to discuss that.

**Comment by HEm3R** [comment]
The same issue with following code:

```java
package test;

public abstract class Test {

	private void test() {
		final List<Entity> ggtJpaEntities = em
				.createQuery("SELECT e FROM Entity e", Entity.class)
				.getResultList();
	}
}
```

and following configuration:

```xml
<module name="Checker">

	<module name="TreeWalker">
		<property name="tabWidth" value="4"/>

		<module name="Indentation">
			<property name="throwsIndent" value="8"/>
			<property name="arrayInitIndent" value="8"/>
		</module>
	</module>
</module>
```

Result:

```
Test.java:7: '.' have incorrect indentation level 4, expected level should be 6.
Test.java:8: '.' have incorrect indentation level 4, expected level should be 6.
```

It was working with CheckStyle 5.8.

**Comment by HEm3R** [comment]
Hi!

The issue is still in 6.6 and it is relevant to using tabs instead of spaces for indentation.

The code snapshot I have provided is working with spaces, but when you replace spaces with tabs and set `<property name="tabWidth" value="4"/>`, indentation error occur (github replaces tabs with spaces, so you have to replace it back yourself).

**Comment by HEm3R** [comment]
And I can confirm it was working in v6.2.

**Comment by rnveach** [comment]
@romani I get no violations in 6.17, with spaces and with tabs.

````
$ cat TestClass.java
package test;

public abstract class Test {

    private void test() {
        final List<Entity> ggtJpaEntities = em
                .createQuery("SELECT e FROM Entity e", Entity.class)
                .getResultList();
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">

    <module name="TreeWalker">
        <property name="tabWidth" value="4"/>

        <module name="Indentation">
            <property name="throwsIndent" value="8"/>
            <property name="arrayInitIndent" value="8"/>
        </module>
    </module>
</module>

$ java -jar checkstyle-6.17-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

**Comment by romani** [comment]
Ok, lets close issue for now. Thanks for testing.
